src/app/(drawerPages)/boosts/BoostsClient.tsx

```diff
@@ -53,25 +56,51 @@ const PageConsts = {
 export default function BoostsPageClient() {
 
   const { address } = useAccount();
+  const loadingCollection = useMemo(() => [null, null, null, null], []);
   const { data: boosts, isLoading } = useBoosts({userAddress: address, gameId: GAME_ID});
   const { claimBoost } = useClaimBoost();
+  const [claimBoostData, setClaimBoostData] = useState<UseMutationResult<AxiosResponse<any, any>, unknown, BoostsClaimData, unknown>>();
+  const queryClient = useQueryClient();
 
-  const boostsCollection = boosts?.filter((boost) => boost.isEnabled).map((boost) => { return {
-    id: boost.id,
-    title: boost.name,
-    description: boost.description,
-    type: boost.boostType,
-    contractAddress: boost.contractAddresses[0],
-    subtitle: '',
-    cta: boost.ctaText,
-    points: boost.points,
-    claimed: boost.claimed,
-    claimable: !boost.claimed,
-    startContent: (
-      iconMapping[boost.icon]
-    ),
-    endContent: <Text>{boost.points.toString()} pts</Text>,
-  }});
+  const boostsCollection =
+    boosts?.filter((boost) => boost.isEnabled).map((boost) => {
+      return {
+        id: boost.id,
+        title: boost.name,
+        description: boost.description,
+        type: boost.boostType,
+        contractAddress: boost.contractAddresses[0],
+        subtitle: '',
+        cta: boost.ctaText,
+        points: boost.points,
+        claimed: boost.claimed,
+        claimable: !boost.claimed,
+        startContent: (
+          iconMapping[boost.icon]
+        ),
+        endContent: <Text>{boost.points.toString()} pts</Text>,
+      }
+    });
+
+  const refetchBoosts = () => {
+    queryClient.invalidateQueries(['boosts', address, GAME_ID]);
+  };
+
+  const resetClaimState = () => {
+    queryClient.invalidateQueries(['boosts', address, GAME_ID]);
+  };
+
+  useEffect(() => {
+    if (claimBoost.isSuccess || claimBoost.isError) {
```

Comment:
would this cause infinite API calls if a call returns an error and we make immediate follow up call?